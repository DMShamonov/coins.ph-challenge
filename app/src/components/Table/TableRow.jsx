// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { TableRowPropsType } from './types.js.flow';

export default function TableRow({
  className,
  children,
}: TableRowPropsType): React.Node {
  return (
    <div className={classnames('table__row', className)}>
      {children}
    </div>
  );
}
