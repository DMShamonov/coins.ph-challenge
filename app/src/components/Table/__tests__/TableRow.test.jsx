import React from 'react';
import { shallow } from 'enzyme';

import TableRow from '../TableRow';

describe('`TableRow` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<TableRow>__CONTENT__</TableRow>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<TableRow className="__CLASSNAME__">__CONTENT__</TableRow>)).toMatchSnapshot();
  });
});
