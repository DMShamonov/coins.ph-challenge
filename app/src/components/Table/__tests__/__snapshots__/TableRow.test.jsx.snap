// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`TableRow\` component should render correctly with \`className\` prop 1`] = `
<div
  className="table__row __CLASSNAME__"
>
  __CONTENT__
</div>
`;

exports[`\`TableRow\` component should render correctly with default props 1`] = `
<div
  className="table__row"
>
  __CONTENT__
</div>
`;
